### PR TITLE
Support is_x11 and is_wayland on EventLoop

### PR DESCRIPTION
**Pull request by dhardy**
- [ ] Tested on all platforms changed
- [x] Added an entry to the `changelog` module if knowledge of this change could be valuable to users
- [ ] Updated documentation to reflect any user-facing changes, including notes of platform-specific behavior
- [ ] Created or updated an example program if it would help users understand this functionality
- [ ] Updated [feature matrix](https://github.com/rust-windowing/winit/blob/master/FEATURES.md), if new features were added or implemented

---

Winit v0.30 does not allow testing `is_wayland` on an `EventLoop` (but only on an `ActiveEventLoop`), though this appears to be possible. This is an issue for me.

The trait names `ActiveEventLoopExtWayland` and `ActiveEventLoopExtX11` are now inaccurate (may cause issues with future additional functionality?), but this is the most straightforward solution.

Fixes https://github.com/rust-windowing/winit/issues/3670

### CODE REVIEW

**Review by kchibisov — CHANGES_REQUESTED** [review]
`EventLoop` is not `Active` it should have its own extensions.

**Comment by dhardy** [comment]
New traits then (a copy of these other than the name)?

**Comment by kchibisov** [comment]
Yes. Which is unfortunate, but it will prevent errors if someone tries to add API for the `EventLoop` which requires `EventLoop` to be actually running.

**Review by notgull — APPROVED** [review]

File: src/changelog/unreleased.md [hunk 1]
```diff
@@ -12,6 +12,7 @@ on how to add them:
 ### Added
 
 - Add `Window::turbo()`, implemented on X11, Wayland, and Web.
+- Implement `ActiveEventLoopExtWayland` and `ActiveEventLoopExtX11` for `EventLoop`.
```
Comment (kchibisov) [inline src/changelog/unreleased.md:15]: This is not correct anymore. It should mention the new traits.

**Review by kchibisov — CHANGES_REQUESTED** [review]

File: src/changelog/unreleased.md (hunk 1 above)
Comment (dhardy) [reply src/changelog/unreleased.md]: Updated

**Review by kchibisov — APPROVED** [review]

**Pull request merged**